### PR TITLE
Clearer rank mismatch error message

### PR DESCRIPTION
**Pull request by RitwikGupta**
`...logits rank (received %s) - 1.`
The `- 1` is confusing for readers, can be mistook for em-dash instead of a subtraction. Also made overall wording more reader-friendly

### CODE REVIEW

**Comment by RitwikGupta** [comment]
I signed it!

**Comment by vincentvanhoucke** [comment]
@tensorflow-jenkins test this please
